libsplinter/src/admin/store/diesel/migrations/postgres/migrations/2020-08-18-213009_admin_service_store/up.sql

```diff
@@ -0,0 +1,136 @@
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS proposed_services_arguments (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    key                       TEXT NOT NULL,
+    value                     TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id),
+    FOREIGN KEY (service_id) REFERENCES proposed_services(service_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS vote_records (
+    circuit_id                TEXT NOT NULL,
+    public_key                BINARY NOT NULL,
+    vote                      TEXT NOT NULL,
+    voter_node_id             TEXT NOT NULL,
+    PRIMARY KEY (circuit_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS circuit_proposal (
+    proposal_type             TEXT NOT NULL,
+    circuit_id                TEXT PRIMARY KEY,
+    circuit_hash              TEXT NOT NULL,
+    requester                 BINARY NOT NULL,
+    requester_node_id         TEXT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS service (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    service_type              TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id),
+    FOREIGN KEY (circuit_id) REFERENCES circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS service_allowed_nodes (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    allowed_node              TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id, allowed_node),
+    FOREIGN KEY (service_id) REFERENCES service(service_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS service_arguments (
+    circuit_id                TEXT NOT NULL,
+    service_id                TEXT NOT NULL,
+    key                       TEXT NOT NULL,
+    value                     TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, service_id),
+    FOREIGN KEY (service_id) REFERENCES service(service_id),
+    FOREIGN KEY (circuit_id) REFERENCES proposed_circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS circuit (
+    circuit_id                TEXT PRIMARY KEY,
+    auth                      TEXT NOT NULL,
+    persistence               TEXT NOT NULL,
+    durability                TEXT NOT NULL,
+    routes                    TEXT NOT NULL,
+    circuit_management_type   TEXT NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS circuit_members (
+    circuit_id                TEXT NOT NULL,
+    member                    TEXT NOT NULL,
+    PRIMARY KEY (circuit_id, member),
+    FOREIGN KEY (circuit_id) REFERENCES circuit(circuit_id) ON DELETE CASCADE
+);
+
+CREATE TABLE IF NOT EXISTS circuit_members_endpoints (
+    member                    TEXT NOT NULL,
```

Comment:
Shouldn't this include the circuit_id?

Reply:
This was initially included just to associate the endpoints with the CircuitNodes to associate the node ids in the circuit's `members` field to their endpoints, but I could see why including the `circuit_id` would be a good idea (if nodes have different endpoints for different circuits)